### ISSUE TITLE
Invalid language attribute values

### ISSUE THREAD

**Issue by axamei** [pulibrary/requests#349]
Violation: Language attribute must have a valid value.

This error came up when testing https://catalog.princeton.edu/requests/9031545?mfhd=8941456&source=pulsearch. The value of the `lang` attribute is not valid. `<dd lang="kor" id="title">Appa rŭl pillyŏ tŭrimnida /</dd>`
 
We should be using the [IANA Language Subtag Registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) to declare the language.

**Comment by sdellis** [comment]
Yes, I agree we should be using the IANA Language Subtag Registry. 

The lang code here is valid ISO 639-2. At the time I coded that, at least, I was having trouble finding information on which codes to use and it seemed that the ISO codes were integrated into the Subtag Registry (apparently only the shortest ISO codes were). Due to the iOS Voice Over bug with languages, I did not have an easy way of testing this.

So, since this code comes from Solr, which comes from Voyager. Since that is the least common denominator for all our apps that deal with catalog data, I think we should map the ISO codes in bibdata (marc_liberation) to the IANA registry subtags. Perhaps this issue should be moved over to the marc_liberation project?

Here is a npm package for mapping the values:
https://github.com/wooorm/iso-639-2

Is there a similar Ruby gem?

**Comment by tampakis** [comment]
Based on: http://accessibility.psu.edu/foreignlanguages/langtaghtml/

Use the two-character code first, then use the three-character code as a backup

**Comment by kevinreiss** [comment]
@tampakis. Has this one advanced at all?

**Comment by christinach** [comment]
@kevinreiss we added a new field to support the IANA Language Subtag registry. https://github.com/pulibrary/marc_liberation/blob/6b7b9e60d65f313fede5a70e5a2cd6e56d634003/marc_to_solr/lib/traject_config.rb#L664.

**Comment by sdellis** [comment]
I don't think we can close this until we use the new field in Requests
